File: TenMinsPy/day58.py
# Check if string contains 8 or 10 digits (even if there's dash, space or + parenthesis) (02) 8-555-6677

import re

class day58:
    def __init__(self):
        maxL1 = 8
        maxL2 = 10
        digit = 0
        s = '(02) 8-555-6677'
        for c in s:
            if c in ('-',' ','+', '+', '(', ')'):
                maxL1 += 1
                maxL2 += 1
            if c.isdigit():
                digit += 1
        if (len(s) == maxL1 or len(s) == maxL2) and (digit == 8 or digit == 10):
            print('nice')
        else:
            print('nah')

day58()

File: TenMinsJava/day58.java
// Check if string contains 8 or 10 digits (even if there's dash, space or + parenthesis) (02) 8-555-6677
package TenMinsJava;

public class day58 {
    public static int digit = 0;
    public static int maxL1 = 8;
    public static int maxL2 = 10;
    public static void main(String[] args) {
        String s = "(02) 8-555-6677";
        for (Character c : s.toCharArray()) {
            if (c == '-' || c == ' ' || c == '+' || c == '(' || c == ')') { maxL1++; maxL2++; }
            if (Character.isDigit(c)) { digit++; }
        }
        if ((s.length() == maxL1 || s.length() == maxL2) && (digit == 8 || digit == 10)) {
            System.err.println("nice");
        }
        else {
            System.err.println("nah");
        }
    }
}
